src/load_inputs/load_dataframe.jl

```diff
@@ -72,6 +129,27 @@ function load_dataframe_from_file(path)
     CSV.read(path, DataFrame, header=1)
 end
 
+function find_matrix_columns_in_dataframe(df::DataFrame,
+        columnprefix::AbstractString;
+        prefixseparator='_')
+    all_columns = names(df)
+
+    # 2 is the length of the '_' connector plus one for indexing
+    get_integer_part(c) = tryparse(Int, c[length(columnprefix)+2:end])
+
+    # if prefix is "ESR", the column name should be like "ESR_1"
+    function is_of_this_column_type(c)
+        startswith(c, columnprefix) &&
+        length(c) >= length(columnprefix) + 2 &&
+        c[length(columnprefix) + 1] == prefixseparator &&
+        !isnothing(get_integer_part(c))
+    end
+
+    columns = filter(is_of_this_column_type, all_columns)
+    columnnumbers = sort!(get_integer_part.(columns))
+    return columnnumbers
+end
+
```

Comment:
This function should perhaps be split into a few functions. If columns like `ESR_1`, `ESR_2` etc are in a dataframe it'll report the column numbers associated with these.